exposed-core/src/main/kotlin/org/jetbrains/exposed/v1/core/statements/api/ExposedDatabaseMetadata.kt

```
@@ -96,9 +92,41 @@ abstract class ExposedDatabaseMetadata(val database: String) {
         else -> this
     }
 
-    /** Returns the normalized column type. */
+    /** Extracts result data about a specific column as [ColumnMetadata]. */
     @InternalApi
-    protected fun normalizedColumnType(columnType: String): String {
+    protected fun RowApi.asColumnMetadata(prefetchedColumnTypes: Map<String, String> = emptyMap()): ColumnMetadata {
+        val defaultDbValue = getObject("COLUMN_DEF", java.lang.String::class.java)?.toString()?.let {
+            sanitizedDefault(it)
+        }
+        val autoIncrement = getObject("IS_AUTOINCREMENT", java.lang.String::class.java)?.toString() == "YES"
+        val type = getObject("DATA_TYPE")?.toString()?.toInt() ?: 0
+        val name = getObject("COLUMN_NAME", java.lang.String::class.java)?.toString() ?: ""
```

Would it be better to have null/error if the name not found? Empty string is still string and another code could try to use it as column name

---

Actually I'm wrong, the original version relied on a (frankly silly) feature for java <-> kotlin. The signature of `getString()` says return type is `String`. But technically, the function will return null if the result is SQL NULL. So yeah, should probably throw an error to mimic this.